src/substrait/type/tests/TypeTest.cpp

```diff
@@ -192,6 +192,14 @@ TEST_F(TypeTest, decodeTest) {
         ASSERT_EQ(typePtr->signature(), "list<str>");
       });
 
+  testDecode<ParameterizedList>(
+      "LIST<any>?",
```

Comment:
Seems question mark can only appear prior to  the position of parameter type , see https://substrait.io/types/type_parsing/#type-syntax-parsing

Reply:
I've removed the decoding enhancement for now.  The intent was to make the decoding more lenient but that should be done as part of a separate PR from this bug fix.